pymatgen/io/vasp/sets.py

```diff
@@ -743,15 +743,16 @@ def __init__(self, structure, prev_incar=None,
     def incar(self):
         incar = super(MPNonSCFSet, self).incar
         if self.prev_incar is not None:
-            incar.update({k: v for k, v in self.prev_incar.items()
-                         if k not in self.kwargs.get("user_incar_settings",
-                                                     {})})
+            incar.update({k: v for k, v in self.prev_incar.items()})
 
         # Overwrite necessary INCAR parameters from previous runs
         incar.update({"IBRION": -1, "ISMEAR": 0, "SIGMA": 0.001,
                       "LCHARG": False, "LORBIT": 11, "LWAVE": False,
                       "NSW": 0, "ISYM": 0, "ICHARG": 11})
 
+        if self.kwargs.get("user_incar_settings", {}) is not None:
```

Comment:
The if statement is not necessary. THe update statement would add nothing if user_incar_settings is not present.